### PR TITLE
Determine valid build file names from buildFileName flag

### PR DESCRIPTION
**Pull request by ains**
Fixes https://github.com/bazelbuild/rules_go/issues/456

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: go/tools/gazelle/gazelle/main.go [hunk 1]
```diff
@@ -35,13 +35,12 @@ import (
...
 
 var (
-	buildFileName  = flag.String("build_file_name", "BUILD", "name of output build files to generate.")
-	buildTags      = flag.String("build_tags", "", "comma-separated list of build tags. If not specified, Gazelle will not\n\tfilter sources with build constraints.")
-	external       = flag.String("external", "external", "external: resolve external packages with new_go_repository\n\tvendored: resolve external packages as packages in vendor/")
-	goPrefix       = flag.String("go_prefix", "", "go_prefix of the target workspace")
-	repoRoot       = flag.String("repo_root", "", "path to a directory which corresponds to go_prefix, otherwise gazelle searches for it.")
-	mode           = flag.String("mode", "fix", "print: prints all of the updated BUILD files\n\tfix: rewrites all of the BUILD files in place\n\tdiff: computes the rewrite but then just does a diff")
-	buildFileNames = []string{"BUILD.bazel", "BUILD"}
+	buildFileName = flag.String("build_file_name", "BUILD,BUILD.bazel", "comma-separated list of valid build file names.\nThe first element of the list is the name of output build files to generate.")
```
Comment (ianthehat) [inline go/tools/gazelle/gazelle/main.go]: Let's reverse this while we are in here and make BUILD.bazel the default?

**Comment by ains** [comment]
Updated @ianthehat although wondering if we want to remove the default
`"build_file_name": attr.string(default="BUILD.bazel,BUILD"),`
As we don't pass the --build_file_name flag to Gazelle if it's empty, and then we don't have the default duplicated

**Comment by ianthehat** [comment]
I almost suggested that, and then I wondered if it was nicer for users of the rules to not have to run the binary to find out what value that attribute would have. But now I think about it even more, I am not sure they should have to care, if we are doing it right nobody is ever going to want to set, or even know about that attribute. Basically I am ambivalent, do whichever you like, I will approve either one as soon as the tests are passing :)

**Comment by ains** [comment]
Ah, we'll leave it for now - test should be fixed :)

**Comment by achew22** [comment]
This change has the affect that when gazelle is run in a directory with `gazelle .` it removes any previous BUILD files and replaces them with BUILD.bazel. Is the objective to get everyone to migrate to BUILD.bazel files? If so could we document that? If not could we switch the default back to 'BUILD' and update the bzl to pass in the flag?

**Comment by jayconrod** [comment]
I don't think the intent of this change was to rename existing files. I'll open an issue and fix it soon, hopefully today.

For new files though, the default will still be BUILD.bazel, which we'd like to encourage. BUILD has too many collisions with existing files and directories.

**Pull request merged**